iam/iam-gcp/src/test/java/com/salesforce/multicloudj/iam/gcp/GcpIamIT.java

```diff
@@ -1,128 +1,619 @@
-			boolean isRecordingEnabled = System.getProperty("record") != null;
-			TransportChannelProvider channelProvider = TestsUtilGcp.getTransportChannelProvider(port);
-			ProjectsSettings.Builder projectsSettingsBuilder = ProjectsSettings.newBuilder()
-					.setTransportChannelProvider(channelProvider);
-			try {
-				if (isRecordingEnabled && useValidCredentials) {
-					projectsClient = ProjectsClient.create(projectsSettingsBuilder.build());
-					IAMSettings.Builder iamSettingsBuilder = IAMSettings.newBuilder();
-					iamClient = IAMClient.create(iamSettingsBuilder.build());
-					return new GcpIam.Builder()
-							.withProjectsClient(projectsClient)
-							.withIamClient(iamClient)
-							.build();
-				} else {
-					GoogleCredentials mockCreds = MockGoogleCredentialsFactory.createMockCredentials();
-					projectsSettingsBuilder.setCredentialsProvider(FixedCredentialsProvider.create(mockCreds));
-					projectsClient = ProjectsClient.create(projectsSettingsBuilder.build());
-					IAMSettings.Builder iamSettingsBuilder = IAMSettings.newBuilder()
-							.setCredentialsProvider(FixedCredentialsProvider.create(mockCreds));
-					iamClient = IAMClient.create(iamSettingsBuilder.build());
-					return new GcpIam.Builder()
-							.withProjectsClient(projectsClient)
-							.withIamClient(iamClient)
-							.build();
-				}
-			} catch (IOException e) {
-				Assertions.fail("Failed to create GCP clients", e);
-				return null;
-			}
-		}
-
-		@Override
-		public String getIdentityName() {
-			return "serviceAccount:chameleon@substrate-sdk-gcp-poc1.iam.gserviceaccount.com";
-		}
-
-		@Override
-		public String getTenantId() {
-			return "projects/substrate-sdk-gcp-poc1";
-		}
-
-		@Override
-		public String getRegion() {
-			return "us-west1";
-		}
-
-		@Override
-		public String getProviderId() {
-			return GcpConstants.PROVIDER_ID;
-		}
-
-		@Override
-		public int getPort() {
-			return port;
-		}
-
-		@Override
-		public List<String> getWiremockExtensions() {
-			return List.of("com.salesforce.multicloudj.iam.gcp.util.IamJsonResponseTransformer");
-		}
-
-		@Override
-		public String getIamEndpoint() {
-			return "https://cloudresourcemanager.googleapis.com";
-		}
-
-
-		@Override
-		public String getTestPolicyEffect() {
-			return "Allow";
-		}
-
-		@Override
-		public List<String> getTestPolicyActions() {
-			return List.of("roles/storage.objectViewer", "roles/storage.objectCreator");
-		}
-
-		@Override
-		public String getTestPolicyName() {
-			return "roles/storage.objectViewer";
-		}
-
-		@Override
-		public void close() {
-			if (projectsClient != null) {
-				projectsClient.close();
-			}
-			if (iamClient != null) {
-				iamClient.close();
-			}
-		}
-	}
 
-}
+    private static final Logger logger = LoggerFactory.getLogger(GcpIamIT.class);
+    private static WireMockServer grpcWireMockServer;
+    private static WireMockGrpcService mockIamService;
+    private static final String GRPC_DIR = "src/test/resources/grpc";
```

Comment:
it's okay to use the same path for recording and let the wiremock framework take care of creating directories. I see now we have to create directories and all manually.

Reply:
scratch this please, I see it's a setup for grpc here https://github.com/wiremock/wiremock-grpc-extension